From: Takashi Iwai <tiwai@suse.de>
Subject: Fix kABI breakage with libertas dev field addition
Patch-mainline: Never, kABI fix
References: bsc#1051510

The patch patches.drivers/libertas-fix-suspend-and-resume-for-SDIO-connected-c
added a new field that breaks kABI.  Since it's a bool, use bit fields
to fold into the previous field.

Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/net/wireless/marvell/libertas/dev.h |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/marvell/libertas/dev.h
+++ b/drivers/net/wireless/marvell/libertas/dev.h
@@ -102,8 +102,12 @@ struct lbs_private {
 	u8 is_polling; /* host has to poll the card irq */
 	u8 fw_ready;
 	u8 surpriseremoved;
+#ifdef __GENKSYMS__
 	u8 setup_fw_on_resume;
-	u8 power_up_on_resume;
+#else
+	u8 setup_fw_on_resume:1;
+	u8 power_up_on_resume:1;
+#endif
 	int (*hw_host_to_card) (struct lbs_private *priv, u8 type, u8 *payload, u16 nb);
 	void (*reset_card) (struct lbs_private *priv);
 	int (*power_save) (struct lbs_private *priv);
